core/controller/block.ts

```diff
@@ -70,7 +70,26 @@ class BlockController extends BaseController {
                 getLastSlotInRound(RoundRepository.getPrevRound());
 
             if (lastBlockInPrevRound >= blockSlot) {
-                await RoundService.backwardProcess();
+                if (System.synchronization) {
+                    let round = RoundRepository.getCurrentRound();
+                    const receivedBlockSlotNumber = SlotService.getSlotNumber(receivedBlock.createdAt);
+                    while (receivedBlockSlotNumber !== round.slots[receivedBlock.generatorPublicKey].slot) {
+                        if (getLastSlotInRound(round) < receivedBlockSlotNumber) {
+                            logger.error(
+                                `[Controller][Block][onReceiveBlock] Impossible to rollback round ` +
+                                `for received block with id: ${receivedBlock.id}, ` +
+                                `height: ${receivedBlock.height}`
+                            );
+                            process.exit(1);
+                        }
+
+                        // backward until we find the right round
+                        RoundService.backwardProcess();
```

Comment:
no await?

Reply:
this is sync func. Need remove async in other parts